src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const indexOfLastElem = this.length - 1;
+    let joinResult = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const element = this[i];
+      const lastElement = (i === indexOfLastElem);
+      const invalidEl = (element === null || element === undefined);
```

Comment:
be consistent, if you have full names in other places, keep them here too
```suggestion
      const isInvalidElement = (element === null || element === undefined);
```